libsrc/osic1p/crt0.s

```diff
@@ -0,0 +1,63 @@
+; ---------------------------------------------------------------------------
+; crt0.s
+; ---------------------------------------------------------------------------
+;
+; Startup code for Ohio Scientific Challenger 1P
+
+.export   _init, _exit
+.import   _main
+
+.export   __STARTUP__ : absolute = 1        ; Mark as startup
+.import   __RAM_START__, __RAM_SIZE__       ; Linker generated
+
+.import    zerobss, initlib, donelib
+
+.include  "zeropage.inc"
+.include  "extzp.inc"
+.include  "osic1p.inc"
+
+; ---------------------------------------------------------------------------
+; Place the startup code in a special segment
+
+.segment  "STARTUP"
+
+; ---------------------------------------------------------------------------
+; A little light 6502 housekeeping
+
+_init:    ldx     #$FF                 ; Initialize stack pointer to $01FF
+          txs
+          cld                          ; Clear decimal mode
+
+; ---------------------------------------------------------------------------
+; Initialize screen width
+; TODO: Can initialization be done in a more idiomatic way?
+; TODO: Create function for changing screen width
+          lda     #$18
+          sta     SCR_LINELEN
```

Comment:
It seems to me as if `SCR_LINELEN` would be a constant. In that case we surely don't want to spend a byte in RAM to load it from but rather use immediate opcodes. If you want to avoid a magic constant in your code you can define a symbol in "your" .inc file - using `=` instead of `:=` to denote that it isn't a (zeropage) address - see Gregs comment.

Note 1: If `SCR_LINELEN` would actually be a variable necessary for conio I'd suggest to create a constructor for it. We don't want to "pollute" crt0.s with initializations for specific C library modules like conio. See https://github.com/cc65/cc65/blob/master/libsrc/apple2/cputc.s for an example of a constructor for conio.